octoprint_bambu_printer/printer/states/paused_state.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import annotations
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from octoprint_bambu_printer.printer.bambu_virtual_printer import (
+        BambuVirtualPrinter,
+    )
+
+import threading
+
+import pybambu.commands
+from octoprint.util import RepeatedTimer
+
+from octoprint_bambu_printer.printer.states.a_printer_state import APrinterState
+
+
+class PausedState(APrinterState):
+
+    def __init__(self, printer: BambuVirtualPrinter) -> None:
+        super().__init__(printer)
+        self._pausedLock = threading.Event()
+
+    def init(self):
+        if not self._pausedLock.is_set():
+            self._pausedLock.set()
+
+        self._printer.sendIO("// action:paused")
+        self._sendPaused()
+
+    def finalize(self):
+        if self._pausedLock.is_set():
+            self._pausedLock.clear()
+
+    def _sendPaused(self):
+        if self._printer.current_print_job is None:
+            self._log.warn("job paused, but no print job available?")
+            return
+        paused_timer = RepeatedTimer(
+            interval=3.0,
+            function=self._printer.report_print_job_status,
+            daemon=True,
+            run_first=True,
+            condition=self._pausedLock.is_set,
+        )
+        paused_timer.start()
+
+    def resume_print(self):

```

Comment:
I think changing this from resume_print to start_new_print will resolve the issue with resuming. With SD card printing the same gcode command (M24) is used for starting and resuming. So that way this function will call the resume command when in paused state, assuming I'm following the refactor logic properly. 

```
    @gcode_executor.register_no_data("M24")
    def _start_print(self):
        self._current_state.start_new_print()
        return True
```